### ISSUE TITLE
Serverless Negs Backends don't work as expected

### ISSUE THREAD

**Issue by nxhuy-github** [terraform-google-modules/terraform-google-lb-http#350]
### TL;DR

In this [code block](https://github.com/terraform-google-modules/terraform-google-lb-http/blob/master/modules/serverless_negs/main.tf#L209C3-L216C4), it must be `lookup(each.value, "description", null)` instead of `lookup(backend.value, "description", null)` !?! The fact that `backend.value` doesn't make any sense in this context I believe !!

### Expected behavior

Use `each.value` instead of `backend.value` in the code block `dynamic "backend" { ... }` of the resource `"google_compute_backend_service"` in the [main.tf](https://github.com/terraform-google-modules/terraform-google-lb-http/blob/master/modules/serverless_negs/main.tf).

### Terraform Configuration

```hcl
dynamic "backend" {
    for_each = toset(each.value["groups"])
    content {
      description = lookup(backend.value, "description", null)
      group       = lookup(backend.value, "group")

    }
}
```

### Terraform Version

```sh
terraform {
  required_version = ">= 1.3"
  required_providers {

    google = {
      source  = "hashicorp/google"
      version = ">= 4.50, < 5.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = ">= 4.50, < 5.0"
    }
    random = {
      source  = "hashicorp/random"
      version = ">= 2.1"
    }
  }

  provider_meta "google" {
    module_name = "blueprints/terraform/terraform-google-lb-http:serverless_negs/v9.1.0"
  }

  provider_meta "google-beta" {
    module_name = "blueprints/terraform/terraform-google-lb-http:serverless_negs/v9.1.0"
  }

}
```

**Comment by imrannayer** [comment]
@nxhuy-github r u getting any error? Both syntaxes r fine for dynamic block. See [TF docs](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks).
